src/__tests__/resources/assetMappings3D.integration.spec.ts

```diff
@@ -0,0 +1,158 @@
+    file = await client.files.upload(fileProps, fileContent);
+    expect(file).toBeTruthy();
+  });
+
+  test(
+    'create revision',
+    async done => {
+      const revisionsToCreate: CreateRevision3D[] = [{ fileId: file.id }];
+      revisions = await retryInSeconds(
+        () => client.revisions3D.create(model.id, revisionsToCreate),
+        3,
+        400,
+        60
+      );
+      expect(revisions.length).toBe(1);
+      done();
+    },
+    10 * 1000
+  );
+
+  const rootAsset = {
+    name: 'test-root',
+    description: 'Root asset for cognitesdk-js test',
+    metadata: {
+      refId: 'test-root',
+    },
+    refId: 'test-root',
+  };
+
+  const childAsset = {
+    name: 'test-child',
+    description: 'Child asset for cognitesdk-js test',
+    metadata: {
+      refId: 'test-child',
+    },
+    parentRefId: 'test-root',
+  };
+
+  test('create assets', async () => {
+    assets = await client.assets.create([childAsset, rootAsset]);
+  });
+
+  let nodes3D: Node3D[];
+  test(
+    'list 3d nodes',
+    async done => {
+      const req = async () =>
+        client.revisions3D
+          .list3DNodes(model.id, revisions[0].id)
+          .autoPagingToArray();
+      nodes3D = await retryInSeconds(req, 5, 404, 5 * 60);
+      expect(nodes3D.map(n => n.name)).toContain('input_Input_1.fbx');
+      done();
+    },
+    5 * 60 * 1000
+  );
+
+  let assetMappingsToCreate: CreateAssetMapping3D[];
+  test('create asset mappings 3d', async () => {
+    assetMappingsToCreate = nodes3D.slice(0, 2).map((node, index) => ({
+      nodeId: node.id,
+      assetId: Number(assets[index].id),
+    }));
+    assetMappings = await client.assetMappings3D.create(
+      model.id,
+      revisions[0].id,
+      assetMappingsToCreate
+    );
+    expect(assetMappings.map(a => a.assetId).sort(simpleCompare)).toEqual(
+      assetMappingsToCreate.map(a => a.assetId).sort(simpleCompare)
+    );
+    expect(assetMappings.length).toBe(2);
+  });
+
+  test('list asset mappings 3d', async () => {
+    const listed = await client.assetMappings3D
+      .list(model.id, revisions[0].id)
+      .autoPagingToArray({ limit: 2 });
+    expect(listed.map(m => m.assetId).sort(simpleCompare)).toEqual(
+      assetMappingsToCreate.map(m => m.assetId).sort(simpleCompare)
+    );
+    expect(listed.map(m => m.nodeId).sort(simpleCompare)).toEqual(
+      assetMappingsToCreate.map(m => m.nodeId).sort(simpleCompare)
+    );
+    expect(listed.length).toBe(2);
+  });
+
+  test('delete asset mappings 3d', async () => {
+    const deleted = await client.assetMappings3D.delete(
+      model.id,
+      revisions[0].id,
+      assetMappingsToCreate
+    );
+    expect(deleted).toEqual({});
+  });
+
+  test('list asset mappings 3d (empty)', async () => {
+    const listed = await client.assetMappings3D
+      .list(model.id, revisions[0].id)
+      .autoPagingToArray();
+    expect(listed).toEqual([]);
+  });
+
+  test('delete revisions', async () => {
```

Comment:
This is good when using [`beforeAll` and `afterAll`](https://jestjs.io/docs/en/setup-teardown#one-time-setup) since then you don't have a test to clean up

Reply:
will use next time then
although, integration test might represent specific use case, so that's why it's good to see wether it works
yeah, right now we don't have any dependencies between DTOs, but what if it we have it later when we decide to go more object oriented?
what if revision fails when you remove it after you attached a file to it (either serverside of here)?

I'd like to test this things anyway... maybe not here tho, but tell your opinion on that please